sklearn/preprocessing/label.py

```diff
@@ -166,6 +166,9 @@ def inverse_transform(self, y):
         """
         self._check_fitted()
 
+        diff = np.setdiff1d(y, np.arange(len(self.classes_)))
+        if diff:
+            raise ValueError("y contains new labels: %s" % str(diff))
```

Comment:
Labels in y should be between 0 and n_classes - 1.
Any value below 0 or above n_classes - 1 does not make sense and cannot be inverted.